src/liballoc/collections/btree/map.rs

```diff
@@ -1606,6 +1641,76 @@ impl<K, V> Clone for Values<'_, K, V> {
     }
 }
 
+/// An iterator produced by calling `drain_filter` on BTreeMap.
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+pub struct DrainFilter<'a, K, V, F>
+where
+    K: 'a + Ord,
```

Comment:
I obeyed the compiler's suggestions, down the cliff I guess. The struct needs it because Drop needs it because for_each needs it because the iterator needs it because the iterator doesn't just hop from handle to handle but needs to search (sometimes). I suppose that we could track the removed handles precisely, or iterate normally and somehow mark or pile up doomed handles for batch removal in the Drop, and the Ord constraint disappears.